flipperdevices/flipperzero-firmware
# [FL-3161] Improved debugging experience for external apps

# What's new

- [FL-3161]  debug: automated support for multiple debug symbol files (incl. applications' plugins)
- trigger a BP when a FAP is loaded when firmware is being debugged - allowing to set breakpoints on newly-loaded code before running it

# Verification 

- Attach debugging session
- Run `./fbt launch_app APPSRC=example_plugins_advanced`
- Breakpoint will trigger; inspect `info files` output
- Set a breakpoint in `applications/examples/example_plugins_advanced/example_advanced_plugins.c`, line 34,
- Resume execution, breakpoint should trigger
- Inspect `info files` again, optionally set a breakpoint in a plugin
- Set bp @ line 47, "return 0"
- Resume execution, breakpoint should trigger
- Check that plugins are unloaded in gdb state

# Checklist (For Reviewer)



[FL-3161]: https://flipperzero.atlassian.net/browse/FL-3161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
